### ISSUE TITLE
[Feature] Filter output of cli commands

### ISSUE THREAD

**Issue by BlackQube** [juanfont/headscale#1974]
### Use case

Some commands, like "headscale nodes list -t" produces a large horizontaly output, that mostly doesn't fit and will be wraped to the next line. This makes it difficult to read.

### Description

It whould be good to be able to filter the output to hide some of the columns.
Currently (without the option "-t, --tags" included) there are the following topics shown:
- ID
- Hostname
- Name 
- MachineKey
- NodeKey
- User
- IP addresses
- Ephemeral
- Last seen
- Expiration
- Connected
- Expired

with "-t, --tags" included:
- ForcedTags
- InvalidTags
- ValidTags

---

I guess, a valid reduced output could be
- ID
- Hostname
- Name 
- User
- IP addresses
- Connected
- Expired

---

The contribution options needed to be checked to submit this request, but I don't think I can be a productive contributor. Sorry...

### Contribution

- [X] I can write the design doc for this feature
- [X] I can contribute this feature

**Comment by idefixcert** [comment]
May I work on refactoring some cli commands that would later make the implementation easier.
As I proposed in [[Feature] cli extension node list #2909](https://github.com/juanfont/headscale/issues/2909)?

**Comment by kradalby** [comment]
Yes please, but keep in mind that we want to change the api in the future so don't invest too much in the grpc part if it is in the way.

**Comment by kradalby** [comment]
I am soliciting feedback for a more coherent CLI design here https://github.com/juanfont/headscale/pull/3375, please comment and help us make the CLI better.
